WordPress/WordPressApi/WordPressOrgXMLRPCApi.swift

```diff
@@ -0,0 +1,183 @@
+     - parameter success:    callback to be called on successful request
+     - parameter failure:    callback to be called on failed request
+
+     - returns:  a NSProgress object that can be used to track the progress of the request and to cancel the request. If the method
+     returns nil it's because something happened on the request serialization and the network request was not started, but the failure callback
+     will be invoked with the error specificing the serialization issues.
+     */
+    public func callMethod(method: String,
+                           parameters: [AnyObject]?,
+                           success: SuccessResponseBlock,
+                           failure: FailureReponseBlock) -> NSProgress?
+    {
+        //Encode request
+        let request: NSURLRequest
+        do {
+            request = try requestWithMethod(method, parameters: parameters)
+        } catch let encodingError as NSError {
+            failure(error: encodingError, httpResponse: nil)
+            return nil
+        }
+
+        // Create task
+        let task = session.dataTaskWithRequest(request) { (data, urlResponse, error) in
+            do {
+                let responseObject = try self.handleResponseWithData(data, urlResponse: urlResponse, error: error)
+                success(responseObject: responseObject, httpResponse: urlResponse as? NSHTTPURLResponse)
+            } catch let error as NSError {
+                failure(error: error, httpResponse: urlResponse as? NSHTTPURLResponse)
+                return
+            }
+        }
+
+        // Progress report
+        let progress = NSProgress()
+        progress.totalUnitCount = 1
+        progress.cancellationHandler = {
+            task.cancel()
+        }
+        task.resume()
+        return progress
+    }
+
+    private func requestWithMethod(method: String, parameters: [AnyObject]?) throws -> NSURLRequest {
+        let mutableRequest = NSMutableURLRequest(URL: endpoint)
+        mutableRequest.HTTPMethod = "POST"
+        mutableRequest.setValue("text/xml", forHTTPHeaderField:"Content-Type")
+        let encoder = WPXMLRPCEncoder(method: method, andParameters: parameters)
+        mutableRequest.HTTPBody = try encoder.dataEncoded()
+
+        return mutableRequest
+    }
+
+    private func handleResponseWithData(data: NSData?, urlResponse:NSURLResponse?, error: NSError?) throws -> AnyObject {
+        guard let data = data,
+            let httpResponse = urlResponse as? NSHTTPURLResponse,
+            let contentType = httpResponse.allHeaderFields["Content-Type"] as? String
+            where error == nil else {
+                if let unwrappedError = error {
+                    throw unwrappedError
+                } else {
+                    throw WordPressOrgXMLRPCApiError.Unknown
+                }
+        }
+
+        if ["application/xml", "text/xml"].filter({ (type) -> Bool in return contentType.hasPrefix(type)}).count == 0 {
```

Comment:
I think the previous code didn't validate the response's content-type. I'm not sure it's a good addition considering how often we come across misconfigured servers. If the response is not XML, the `WPXMLRPCDecoder` will fail anyway.